debian/packages.d/systemd.customise.add/init.d/20confd.sh

```diff
@@ -0,0 +1,39 @@
+# Apply configurations from conf.d
+#
+# Copyright (C) 2018 Benedict Lau <benedict.lau@groundupworks.com>
+
+CONFDIR=conf.d
+
+try_partition() {
+    # Create file system node from partition
+    #mknod /dev/$1 b $2 $3 2>/dev/null
+
+    # Mount partition as read-only
+    mount /dev/$1 /mnt -o ro
+
+    # Find conf.d in root of partition and extract in alphabetical order each
+    # tar.gz archive into the ramdisk /etc, overwriting any existing file
+    if [ -d "/mnt/$CONFDIR" ]; then
+        confs=$(find /mnt/$CONFDIR -name *.tar.gz)
+        for conf in $confs; do
+            echo Applying configurations from /dev/$1: $conf
+            tar --extract --gzip -f $conf -C /etc
+        done
+    fi
+
+    # Unmount partition
+    umount /mnt
+}
+
+# Mount proc if not already mounted
+#mount -t proc proc /proc
+
+cat /proc/partitions | while read major minor size name; do
```

Comment:
This reads `conf.d` from all partitions. I guess it's okay for now? It'll be a little weird if multiple partitions have configurations because tar archive order rule doesn't apply here cross partitions.

Reply:
we could exit on first found - which has issues if the files are hard to overwrite.  we could also collate all the tar files from all the partitions - but that has an unwanted increase in complexity.

by checking all partitions, we try to avoid painting ourselves into an unexpected corner

Reply:
Neither first found or collect sort seem better than current. They are just different but just as non-ideal. I suggest keeping it as is and discourage people from having two partitions with conf.d at root, and state current behaviour.

Reply:
I suggested the collect-and-sort method as that is how a lot of things (like systemd) handle multiple sources of config overrides - its an annoying amount of additional work, but could be done if this ever becomes an issue

Reply:
On WINDOWS machines, only the FIRST partition is ever recognized  OS on usb devices.

Unless you have some reason to put config on partition >2 you could just stick with partition 1 to make it most compatible.

Reply:
The issue still exists when there are multiple physical disks connected to the board